src/DotNetWorker.Core/IFunctionsApplication.cs

```diff
@@ -12,5 +14,18 @@ internal interface IFunctionsApplication
         Task InvokeFunctionAsync(FunctionContext context);
 
         FunctionContext CreateContext(IInvocationFeatures features);
+
+        /// <summary>
+        /// Cancels an invocations associated <see cref="CancellationTokenSource"/>.
+        /// </summary>
+        /// <param name="invocationId">Invocation ID</param>
+        void CancelInvocation(string invocationId);
+
+        /// <summary>
+        /// Removes a given invocation from the <see cref="FunctionInvocationDictionary"/>.
+        /// Should be called on completed invocations.
+        /// </summary>
+        /// <param name="invocationId">Invocation ID</param>
+        void RemoveInvocationRecord(string invocationId);
```

Comment:
naming nit: Wonder we should be more generic here instead of very specific about removing the record. Something like "Complete" ?
This comment could be totally not valid as well.

Reply:
Something like `CompleteInvocation`? Could we be doing something else in here so that we need to make this more generic?

Reply:
My other comment mentions an idea to remove the need for this logic by shifting the responsibility to the caller. But anyways - for cases like this I like using `IDisposable`. So whatever "Start" method we have would return a disposable object, and the dispose would invoke the pairing `Complete` operation. This makes it easy to use via `using` statements and is also a pattern devs will be familiar with.